samza-core/src/main/java/org/apache/samza/clustermanager/HostAwareContainerAllocator.java

```diff
@@ -81,13 +87,27 @@ public void assignResourceRequests()  {
 
         if (expired) {
           updateExpiryMetrics(request);
-          if (resourceAvailableOnAnyHost) {
-            log.info("Request for container: {} on {} has expired. Running on ANY_HOST", request.getContainerID(), request.getPreferredHost());
-            runStreamProcessor(request, ResourceRequestState.ANY_HOST);
+
+          if (standbyContainerManager.isPresent()) {
+
+            // if standby is enabled and an alternative-anyhost-resource is available, we try to use it
+            if (resourceAvailableOnAnyHost) {
+              standbyContainerManager.get().handleExpiredResourceRequest(containerID, request, Optional.of(peekAllocatedResource(ResourceRequestState.ANY_HOST)), this, resourceRequestState);
+            } else {
+              standbyContainerManager.get().handleExpiredResourceRequest(containerID, request, Optional.empty(), this, resourceRequestState);
+            }
+
           } else {
-            log.info("Request for container: {} on {} has expired. Requesting additional resources on ANY_HOST.", request.getContainerID(), request.getPreferredHost());
-            resourceRequestState.cancelResourceRequest(request);
-            requestResource(containerID, ResourceRequestState.ANY_HOST);
+
+            if (resourceAvailableOnAnyHost) {
```

Comment:
prefer consolidating resourceAvailableOnAnyHost checks instead of having them checked at multiple places

Reply:
This separation keeps the path taken in case standbyContainerManager is and isnt present clear.